### PR TITLE
Astc tests

### PR DESCRIPTION
**Pull request by abbaswasim**
Adds tests for newly added encode --format astc_format

I had to add more astc input ktx2 files because I needed these from a known source.

### CODE REVIEW

File: clitests/input/ktx2/valid_R8G8B8_UNORM_PACK32_2D.ktx2
Comment (lexaknyazev) [inline clitests/input/ktx2/valid_R8G8B8_UNORM_PACK32_2D.ktx2:1]: The file name seems incorrect. Its format is `VK_FORMAT_R8G8B8_UNORM`.

File: clitests/tests/encode/encode_astc.json [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{
+    "description": "Test default behavior of UASTC encoding.",
```
Comment (lexaknyazev) [inline clitests/tests/encode/encode_astc.json:2]: The description should say ASTC.

File: clitests/input/ktx2/valid_R8G8B8_UNORM_PACK32_2D.ktx2
Comment (abbaswasim) [reply clitests/input/ktx2/valid_R8G8B8_UNORM_PACK32_2D.ktx2:1]: fixed in https://github.com/KhronosGroup/KTX-Software-CTS/pull/33/commits/bb7cdf30e166ecfed1ec605289073f7a7eec471b

File: clitests/tests/encode/encode_astc.json (hunk 1 above)
Comment (abbaswasim) [reply clitests/tests/encode/encode_astc.json]: Fixed in https://github.com/KhronosGroup/KTX-Software-CTS/pull/33/commits/bb7cdf30e166ecfed1ec605289073f7a7eec471b

File: clitests/input/ktx2/valid_R8G8B8_UNORM_PACK32_2D.ktx2
Comment (abbaswasim) [reply clitests/input/ktx2/valid_R8G8B8_UNORM_PACK32_2D.ktx2:1]: yes, I will remove the pack32.

Comment (abbaswasim) [reply clitests/input/ktx2/valid_R8G8B8_UNORM_PACK32_2D.ktx2:1]: Fixed in https://github.com/KhronosGroup/KTX-Software-CTS/pull/33/commits/532198fe61559d6a34adeacc455ce21f58cd1994

File: clitests/tests/encode/encode_astc_bad_format.json [hunk 2]
```diff
@@ -0,0 +1,258 @@
+{
+    "description": "Test encode command with unsupported formats.",
+    "command": "ktx encode --testrun --threads 1 --format ${format} input/ktx2/valid_known_R8G8B8_UNORM_2D.ktx2 output/encode/encode_error_format/output_${format}.ktx2",
+    "status": 1,
+    "stderr": "golden/encode/encode_error_format/common_format.err.txt",
+    "args": {
+        "format": [
+            "R4G4_UNORM_PACK8",
```
Comment (MarkCallow) [inline clitests/tests/encode/encode_astc_bad_format.json:8]: I think we should include some ASTC formats here to make sure people can't decode and re-encode differently.

Actually a different error message would be good in this case. Something like "already encoded as ASTC.". So testing those formats will likely be a different test.

Comment (wasimabbas-arm) [reply clitests/tests/encode/encode_astc_bad_format.json:8]: This test is checking for invalid `--format`s to `ktx encode` in particular. ASTC format to `--format` in not invalid. As you said checking existing format already ASTC is a different error and also requires different functionality first. I shall add that first.

**Comment by wasimabbas-arm** [comment]
> What are the `[clitests/input/ktx2/valid_ASTC_4x4_SRGB_BLOCK_known_R8G8B8_2D.ktx2` files for? They don't appear to be used in any test.

As I mentioned in the PR description:

> I had to add more astc input ktx2 files because I needed these from a known source.

These confused me a bit I now remember I only had to generated those using one of the tests which I then copied into `golden/encode/encode_astc`. I will remove those. There were pushed by mistake. These should be the exact copy of the golden ones.

File: clitests/tests/encode/encode_astc_bad_format.json (hunk 2 above)
Comment (MarkCallow) [reply clitests/tests/encode/encode_astc_bad_format.json:8]: Okay. Add the ASTC tests in another test.

**Pull request merged**